mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingFeature.kt

```diff
@@ -0,0 +1,118 @@
+package com.mineinabyss.armorstandlock
+
+import com.mineinabyss.components.armorstandlock.LockArmorStand
+import com.mineinabyss.components.playerData
+import com.mineinabyss.geary.minecraft.access.toGeary
+import com.mineinabyss.idofront.commands.arguments.stringArg
+import com.mineinabyss.idofront.commands.extensions.actions.playerAction
+import com.mineinabyss.idofront.entities.toPlayer
+import com.mineinabyss.idofront.messaging.error
+import com.mineinabyss.idofront.messaging.info
+import com.mineinabyss.idofront.messaging.success
+import com.mineinabyss.idofront.plugin.registerEvents
+import com.mineinabyss.mineinabyss.core.AbyssFeature
+import com.mineinabyss.mineinabyss.core.MineInAbyssPlugin
+import com.mineinabyss.mineinabyss.core.commands
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.bukkit.Bukkit
+
+@Serializable
+@SerialName("lock_armorstand")
+class ArmorStandLockingFeature : AbyssFeature {
+    override fun MineInAbyssPlugin.enableFeature() {
+        registerEvents(ArmorStandLockingListener())
+
+        commands {
+            mineinabyss {
+                "lock"(desc = "Protection related commands") {
+                    "toggle"(desc = "Toggles if an armor stand should be protected or not") {
+                        playerAction {
+                            val entity = player.playerData.recentRightclickedEntity
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+
+                            locked.lockState = !locked.lockState
+                            if (locked.lockState) player.success("This armor stand is now protected!")
+                            else player.error("This armor stand is no longer protected!")
+                        }
+                    }
+                    "add"(desc = "Add a player to this armor stand.") {
+                        val playerName by stringArg()
+                        playerAction {
+                            val entity = player.playerData.recentRightclickedEntity
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+                            val uuid = Bukkit.getOfflinePlayer(playerName).uniqueId
+
+                            if (locked.allowedAccess.contains(uuid)) {
+                                player.error("This player can already interact with this armor stand")
+                                return@playerAction
+                            }
+                            else {
+                                locked.allowedAccess += uuid
```

Comment:
allowedAccess is an immutable list, im pretty sure this is creating a new one under the hood (you should have gotten a warning in the IDE iirc). just making it mutable should be fine.